### DISCUSSION TITLE
Brew uninstall node potentially leaving behind files?

### DISCUSSION THREAD

**Discussion by youn123** [Homebrew/discussions#3663] [Getting started]
Hello,

I recently upgraded to Node 16.17.0 using the installer on the official website [https://nodejs.org/en](https://nodejs.org/en/), without realizing I've installed Node using homebrew previously (16.13.0). The installer put the version of node I want under `/usr/local/bin` while my homebrew installation was under `/opt/homebrew/opt/node@16/bin/node`.

So I did `brew uninstall node@16` which seems to have uninstalled node successfully, and now I am able to pick up the new node version:

```
node -v # v16.17.0
which node # /usr/local/bin/node
```

However it seems like I am still picking up npm under homebrew

```
which npm # /opt/homebrew/bin/npm
```

There still exists `/opt/homebrew/bin/npm` and `/opt/homebrew/bin/npx`, so I was wondering what their relation to the `node@16` installation was (E.g. if they were supposed to be removed with the uninstall) and what the correct/safe way to start picking up `/usr/local/bin/npm` is. Let me know if this is better served as an issue. Thank you.

**Comment by Bo98** [comment]
You might have Node 18 installed. Does `brew uninstall node` work? If not what does `ls -l /opt/homebrew/bin/npm` output?

**Reply by youn123** [reply]
Hey Bo thanks for your reply.

`brew uninstall node` doesn't work (`Error: No such keg: /opt/homebrew/Cellar/node`). 2nd one outputs:

```
lrwxr-xr-x  1 youn  admin  49 Dec 13  2021 /opt/homebrew/bin/npm -> /opt/homebrew/lib/node_modules/npm/bin/npm-cli.js
```

**Reply by gogoshanghai** [reply]
having the same problem... did you fix this??

**Reply by bluec** [reply]
And me. Any updates?

**Reply by bluec** [reply]
FWIW I did this...

```
rm -rf /opt/homebrew/bin/npm
rm -rf /opt/homebrew/bin/npx
rm -rf /opt/homebrew/lib/node_modules/npm/
```

It allowed me to get my native nvm versions of npm working. No idea yet whether it has boshed anything in homebrew but there was nothing node related in the dependency tree
